src/seahawk_deck/seahawk_deck/motion_controller.py

```diff
@@ -18,43 +19,119 @@ def __init__(self):
         """Initialize this node"""
         super().__init__('motion_controller')
         self.twist_pub = self.create_publisher(Twist, 'drive/twist', 10)
-        self.motor_pub = self.create_publisher(Int16MultiArray, 'drive/motors', 10) # subscription to this in rov/i2c_proxy
+        self.motor_pub = self.create_publisher(Float32MultiArray, 'drive/motors', 10)
         self.subscription = self.create_subscription(Joy, 'joy', self._callback, 10)
+    
+    def thrust_control(self, direction1:float, direction2:float)->float:
```

Comment:
I mentioned a readability problem with the variable names. The problem was that the names didn't mean anything. Calling them "direction#" leaves them without any meaning, still.

It would also be good to explain what this code does, and how it does it. The comment claims that it restricts the range to [-1, 1], but the code definitely doesn't.

Reply:
I agree the functions need names that describe what they do / are. And / or proper doc strings that do such.

Code is only readable to you right now because you stared at it for a few hours straight.

Needs more documentation

Reply:
I dont want another Carter situation on our hands

Reply:
A   w h a t

Reply:
@MegaMichaelXP Could you add some code comments to do this?